pkg/cmd/plan.go

```diff
@@ -0,0 +1,276 @@
+// TODO (cory) Make these specific to the target lang?
+type templateVars struct {
+	Name   string
+	Module string
+}
+
+func createCommand(c *cli.Context) error {
+	if c.Args().Len() != 1 {
+		return errors.New("this command requires one argument -- specify the plan name")
+	}
+
+	cfg := &config.EnvConfig{}
+	if err := cfg.Load(); err != nil {
+		return err
+	}
+
+	plan_name := c.String("plan")
+	target_lang := c.String("target")
+	remote := c.String("remote")
+	module := c.String("module")
+
+	pdir := filepath.Join(cfg.Dirs().Plans(), plan_name)
+	repo, err := git.PlainInit(pdir, false)
+	if err != nil {
+		return err
+	}
+
+	if remote != "" {
+		_, err := repo.CreateRemote(&gitcfg.RemoteConfig{
+			Name: "origin",
+			URLs: []string{remote},
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Get file templates for the supplied target lang
+	asset_path := fmt.Sprintf("/%s-templates", target_lang)
+	var tset ttmpl.TemplateSet
+	err = ttmpl.Fill(asset_path, &tset)
+	if err != nil {
+		return err
+	}
+
+	if tset == nil {
+		return fmt.Errorf("unknown language target %s", target_lang)
+	}
+
+	for _, ts := range tset {
+		tmpl, err := template.New(ts.Filename).Parse(ts.Template)
+		if err != nil {
+			return err
+		}
+		f, err := os.Create(filepath.Join(pdir, tmpl.Name()))
+		if err != nil {
+			return err
+		}
+		err = tmpl.Execute(f, templateVars{Name: plan_name, Module: module})
+		if err != nil {
+			return err
+		}
+		f.Close()
+	}
+	return nil
+}
+
+func importCommand(c *cli.Context) error {
+	cfg := &config.EnvConfig{}
+	if err := cfg.Load(); err != nil {
+		return err
+	}
+
+	source := c.String("source")
+
+	parsed, err := url.Parse(source)
+	if err != nil {
+		return err
+	}
+
+	// determine the destation, either from flag or intuited from the soruce.
+	baseDest := c.String("name")
+	if baseDest == "" {
+		baseDest = filepath.Base(parsed.Path)
+	}
+	dstPath := filepath.Join(cfg.Dirs().Home(), "plans", baseDest)
+
+	// Use git to clone. Any scheme supported by git is acceptable.
+	if c.Bool("git") {
+		return clonePlan(dstPath, source)
+	}
+
+	// not using git, simply symlink the directory. Remove the file:// scheme if it is included.
+	var srcPath string
+	switch parsed.Scheme {
+	case "file":
+		srcPath = parsed.Path
+	case "":
+		srcPath = source
+	default:
+		return fmt.Errorf("unknown scheme %s for local files. did you forget to pass --git?", parsed.Scheme)
+	}
+	return symlinkPlan(dstPath, srcPath)
+}
+
+func symlinkPlan(dst, src string) error {
+	abs, err := filepath.Abs(src)
+	if err != nil {
+		return err
+	}
+	ev, err := filepath.EvalSymlinks(abs)
+	if err != nil {
+		return err
+	}
+	return os.Symlink(ev, dst)
+}
+
+func clonePlan(dst, src string) error {
+
+	cloneOpts := git.CloneOptions{
+		URL:      src,
+		Progress: os.Stderr,
+	}
+
+	_, err := git.PlainClone(dst, false, &cloneOpts)
+	return err
+}
+
+func rmCommand(c *cli.Context) error {
```

Comment:
To be consistent with the rest of the CLI, this should take a `--name` flag instead of expecting the plan as an argument. BTW -- I like the `--yes` UX :-)

Reply:
And it should print a confirmation message:

```
plan at <full_path> removed.
```

Reply:
Message added